blocks/filter-group/filter-group.js

```diff
@@ -1,42 +1,193 @@
+) || [];
+
+/**
+ * TO-DO: remove this when ideas page content and functionality is finalized.
+ * 
+ * It's intended for data-tags to be applied to cards of dynamically loaded ideas articles.
+ * This mocks up what tags are applied per placeholder card element on the Ideas page.
+ */
+const applyTemporaryMockupTags = () => {
+    if (window.location.pathname != '/ideas/') return;
+
+    const mockupTagsAppliedToCards = [
+        'Design Research,Generative AI Design',
+        'Design Systems,Design Process',
+        'Design Systems,Design Leadership',
+        'Design Strategy',
+        'Generative AI Design',
+    ];
+
+    document.querySelectorAll('.card')?.forEach((card, i) => {
+        if (mockupTagsAppliedToCards.length < i) {
+            card.dataset.tags = 'Design Stategy';
+            return;
+        }
+        card.dataset.tags = mockupTagsAppliedToCards[i];
+    });
+}
+
+/**
+ * TO-DO: work with real Ideas page content. This function is currently mockup/placeholder functionality
+ * and will likely need to be moved to the ideas/stories feed block.
+ * 
+ * Update all articles displayed on the page with ones that have tags
+ * that match the currently selected filters.
+ */
+const refreshArticleContent = (selectedFilters = []) => {
+    // TO-DO: remove this when ideas page content and functionality is finalized.
+    applyTemporaryMockupTags();
+
+    // Show/hide article cards based on selected tags.
+    document.querySelectorAll('.card')?.forEach(article => {
+        // "All" reset. Display all articles.
+        if (selectedFilters.length == 0){
+            article.style.display = '';
+            return;
+        }
+        
+        // Show/hide based on data-tags applied to the article.
+        if (article.dataset?.tags) {
+            const tagsArray = article.dataset.tags.split(',');
+            const hasTag = selectedFilters.some(tag => tagsArray.includes(tag));
+            article.style.display = hasTag ? '' : 'none';
+        }
+    });
+};
+
+/**
+ * Handle click on a filter button from the parent container. Events are bubbled
+ * up from the button. Triggers functionality to update filtered content and
+ * selected state.
+ * @param {PointerEvent} event
+ */
+const handleFilterClick = function (event) {
+    const filterGroup = this;
+    const clickedButton = event.target.closest('.filter-group__button');
+    if (!clickedButton) return;
+
+    // State of button and whether it's the special "All" filter reset.
+    const isSelected = clickedButton.getAttribute('aria-pressed') == 'true';
+    const isAllFilter = clickedButton.matches(':first-of-type');
+
+    if (isAllFilter) {
+        // "All" filter -----
+        // Does nothing if already selected.
+        if (isSelected) return;
+        // Set all other filters as unselected when "All" is selected.
+        filterGroup.querySelectorAll('.filter-group__button').forEach((btn) => {
```

Comment:
can we add `.filter-group__button` to a constant to pull the queries out of the function?

We seem to be only calling them once but to me it is more readable to set up all the calls at the top of the file, more like you're doing with `isAllFilter`

Reply:
Good idea. I've updated this to bring these selectors and list of elements to the top. [0915d63](https://github.com/adobe/adobe-design-website/pull/81/commits/0915d637e9b7ee2411477f17788b326c5d67d1f4)